### PR TITLE
Feature/pdpdevtool 4541 vscode update instances

### PR DESCRIPTION
**Pull request by antoniofruci**

### CODE REVIEW

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 1]
```diff
@@ -41,9 +43,31 @@ export default class UpdateObject extends BaseAction {
...
+		const yes = this.translationService.getMessage(ANSWERS.YES);
+		const no = this.translationService.getMessage(ANSWERS.NO);
+		const includeInstancesAnswer = await window.showQuickPick([yes, no], {
+			placeHolder: this.translationService.getMessage(UPDATE_OBJECT.QUESTIONS.INCLUDE_INSTANCES),
+			canPickMany: false,
+		});
+
+
+		let includeInstancesFlag = '';
+		if (includeInstancesAnswer === this.translationService.getMessage(ANSWERS.YES)) {
```
Comment (lfpvillegas) [inline packages/vscode-extension/src/commands/UpdateObject.ts:55]: You could use previously defined constant.

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 2]
```diff
@@ -41,9 +43,31 @@ export default class UpdateObject extends BaseAction {
...
+			const overwriteInstances = await window.showQuickPick([continueMessage, cancelMessage], {
+				placeHolder: this.translationService.getMessage(UPDATE_OBJECT.OVERWRITE_INSTANCES),
+				canPickMany: false,
+			});
+	
+			if (!overwriteInstances || overwriteInstances === this.translationService.getMessage(ANSWERS.CANCEL)) {
```
Comment (lfpvillegas) [inline packages/vscode-extension/src/commands/UpdateObject.ts:61]: you could use previously defined constant cancel (also in line 41)

**Review by lfpvillegas — CHANGES_REQUESTED** [review]

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 3]
```diff
@@ -31,19 +33,42 @@ export default class UpdateObject extends BaseAction {
...
-			this.messageService.showInformationMessage(this.translationService.getMessage(UPDATE_OBJECT.PROCESS_CANCELED));
+		if (!overwriteObject || overwriteObject === cancelMessage) {
+			this.messageService.showInformationMessage(processCanceledMessage);
 			return;
 		}
 
+		const includeInstancesAnswer = await window.showQuickPick([yes, no], {
+			placeHolder: this.translationService.getMessage(UPDATE_OBJECT.QUESTIONS.INCLUDE_INSTANCES),
+			canPickMany: false,
+		});
```
Comment (davidecorreu) [inline packages/vscode-extension/src/commands/UpdateObject.ts:53]: This should only be asked if the selected object is a custom record. There is some logic missing here.

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 4]
```diff
@@ -31,19 +33,42 @@ export default class UpdateObject extends BaseAction {
...
+
+		let includeInstancesFlag = '';
+		if (includeInstancesAnswer === yes) {
+			const overwriteInstances = await window.showQuickPick([continueMessage, cancelMessage], {
```
Comment (davidecorreu) [inline packages/vscode-extension/src/commands/UpdateObject.ts:57]: I'm not in favor of asking twice about overwriting, once for regular objects, and again for the instances of custom records. I think is a bad user experience for the developer. I would prefer to ask only once, either overwrite objects or overwrite objects and its instances.
I'm actually not sure that we could update more than one object at a time, but I would like to be only be asked about overwriting only once in any case.

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 5]
```diff
@@ -103,4 +99,8 @@ export default class UpdateObject extends BaseAction {
 			return this.unsuccessfulValidation(e.getErrorMessage());
 		}
 	}
+
+	private static isCustomRecord(scriptid: string): boolean {
```
Comment (lfpvillegas) [inline packages/vscode-extension/src/commands/UpdateObject.ts:103]: we don't really need that this method is static, it is not called for anywhere else.

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 6]
```diff
@@ -31,19 +34,36 @@ export default class UpdateObject extends BaseAction {
...
+
+		if (UpdateObject.isCustomRecord(scriptId)) {
+			const includeInstancesAnswer = await window.showQuickPick([yes, no], {
+				placeHolder: this.translationService.getMessage(UPDATE_OBJECT.QUESTIONS.INCLUDE_INSTANCES),
+				canPickMany: false,
+			});
+			if (includeInstancesAnswer === yes) {
+				suiteCloudCommand.includeinstances = INCLUDE_INSTANCES;
+				overwriteMessage = this.translationService.getMessage(UPDATE_OBJECT.OVERWRITE_INSTANCES, scriptId);
+			}
```
Comment (davidecorreu) [inline packages/vscode-extension/src/commands/UpdateObject.ts:51]: I think we are missing the case were the user doesn't respond to the question or just hit ESC , in that case I think we should stop the command flow. Something like:
```
if (!includeInstancesAnswer) {
  return;
}
```

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 7]
```diff
@@ -31,19 +34,36 @@ export default class UpdateObject extends BaseAction {
 
 		const continueMessage = this.translationService.getMessage(ANSWERS.CONTINUE);
 		const cancelMessage = this.translationService.getMessage(ANSWERS.CANCEL);
-		const override = await window.showQuickPick([continueMessage, cancelMessage], {
-			placeHolder: this.translationService.getMessage(UPDATE_OBJECT.OVERRIDE, scriptId),
+		const yes = this.translationService.getMessage(ANSWERS.YES);
+		const no = this.translationService.getMessage(ANSWERS.NO);
+		const processCanceledMessage = this.translationService.getMessage(UPDATE_OBJECT.PROCESS_CANCELED);
+		const suiteCloudCommand = { scriptid: [scriptId], includeinstances: '' };
```
Comment (davidecorreu) [inline packages/vscode-extension/src/commands/UpdateObject.ts:40]: I would prefer to name this one `suiteCloudCommandOptions` / `commandOptions` or similar one with "options" in the name.

File: packages/vscode-extension/src/commands/UpdateObject.ts [hunk 8]
```diff
@@ -31,19 +34,39 @@ export default class UpdateObject extends BaseAction {
...
+				placeHolder: this.translationService.getMessage(UPDATE_OBJECT.QUESTIONS.INCLUDE_INSTANCES),
+				canPickMany: false,
+			});
+			if (includeInstancesAnswer === yes) {
+				suiteCloudCommandOptions.includeinstances = INCLUDE_INSTANCES;
+				overwriteMessage = this.translationService.getMessage(UPDATE_OBJECT.OVERWRITE_INSTANCES, scriptId);
+			} else if(!includeInstancesAnswer) {
+				this.messageService.showInformationMessage(processCanceledMessage);
+				return;
+			}
```
[the remark this answers is not in the document]
Comment (antoniofruci) [reply packages/vscode-extension/src/commands/UpdateObject.ts:54]: True, I'll move it

**Review by lfpvillegas — APPROVED** [review]

**Pull request merged**